test/integration/setup.js

```diff
@@ -121,33 +147,40 @@ async function setup() {
   // trigger initialize and update master address
   await disposableMCR.initializeNextMcr(mc.address, master.address);
 
+  const priceFeedOracleAssets = [
+    { contract: dai, aggregator: chainlinkDAI, decimals: 18 },
+    { contract: stETH, aggregator: chainlinkSteth, decimals: 18 },
+    { contract: rETH, aggregator: chainlinkReth, decimals: 18 },
+    { contract: aWETH, aggregator: chainlinkAweth, decimals: 18 },
+    { contract: st, aggregator: chainlinkSt, decimals: 18 },
+    { contract: enzymeVault, aggregator: chainlinkEnzymeVault, decimals: 18 },
+    { contract: usdc, aggregator: chainlinkUSDC, decimals: usdcDecimals },
+    { contract: debtUsdc, aggregator: chainlinkUSDC, decimals: debtUsdcDecimals },
+  ];
   const priceFeedOracle = await ethers.deployContract('PriceFeedOracle', [
-    [dai, stETH, usdc, enzymeVault].map(c => c.address),
-    [chainlinkDAI, chainlinkSteth, chainlinkUSDC, chainlinkEnzymeVault].map(c => c.address),
-    [18, 18, 6, 18],
+    priceFeedOracleAssets.map(a => a.contract.address),
+    priceFeedOracleAssets.map(a => a.aggregator.address),
+    priceFeedOracleAssets.map(a => a.decimals),
   ]);
 
-  // placeholder is swapped with the actual one after master is initialized
-  const swapOperatorPlaceholder = { address: AddressZero };
-
-  const legacyPool = await ethers.deployContract(
-    'LegacyPool',
-    [master, priceFeedOracle, swapOperatorPlaceholder, dai, stETH, enzymeVault, tk].map(c => c.address),
-  );
```

Comment:
does anyone know why we have the `swapOperatorPlaceholder`? all tests are passing by passing in the `swapOperator` directly.

I also found out that the pool swapOperator was still the zero address. so i elected to remove it all together to simplify the code